src/test/java/com/artipie/rpm/http/RpmRepoSliceITCase.java

```diff
@@ -0,0 +1,180 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.rpm.http;
+
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import com.artipie.asto.memory.InMemoryStorage;
+import com.artipie.http.auth.BasicIdentities;
+import com.artipie.http.auth.Identities;
+import com.artipie.http.auth.Permissions;
+import com.artipie.http.slice.LoggingSlice;
+import com.artipie.rpm.Digest;
+import com.artipie.rpm.NamingPolicy;
+import com.artipie.rpm.Rpm;
+import com.artipie.rpm.TestRpm;
+import com.artipie.vertx.VertxSliceServer;
+import io.vertx.reactivex.core.Vertx;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import org.cactoos.list.ListOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.text.StringContainsInOrder;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.containers.GenericContainer;
+
+/**
+ * Test for {@link RpmSlice}.
+ * @since 0.11
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public final class RpmRepoSliceITCase {
+
+    /**
+     * Vertx instance.
+     */
+    private static final Vertx VERTX = Vertx.vertx();
+
+    /**
+     * Settings files.
+     */
+    private final Path stngs = Paths.get("src/test/resources/example.repo");
+
+    /**
+     * Vertx slice server instance.
+     */
+    private VertxSliceServer server;
+
+    /**
+     * Container.
+     */
+    private GenericContainer<?> cntn;
+
+    @Test
+    void listAndInstall() throws Exception {
+        this.start(Permissions.FREE, Identities.ANONYMOUS, "");
+        this.asserts();
+    }
+
+    @Test
+    void listAndInstallWithAuth() throws Exception {
+        final String mark = "mark";
+        final String pswd = "abc";
+        this.start(
+            (name, perm) -> mark.equals(name) && "download".equals(perm),
+            new BasicIdentities(
+                (name, pass) -> {
+                    final Optional<String> res;
+                    if (mark.equals(name) && pswd.equals(pass)) {
+                        res = Optional.of(name);
+                    } else {
+                        res = Optional.empty();
+                    }
+                    return res;
+                }
+            ),
+            String.format("%s:%s@", mark, pswd)
+        );
+        this.asserts();
+    }
+
+    @AfterEach
+    void stopContainer() throws IOException {
+        this.server.close();
+        this.cntn.stop();
+        Files.deleteIfExists(this.stngs);
+    }
+
+    @AfterAll
+    static void close() {
+        RpmRepoSliceITCase.VERTX.close();
+    }
+
+    /**
+     * Assertions.
+     * @throws Exception On error
+     */
+    private void asserts() throws Exception {
```

Comment:
@olenagerasimova I think it would be better, from test readability point of view, to have separate methods for executing commands in the container and separate for assertions.
If we look at this test body it's not clear what is being tested:
```java
this.start(Permissions.FREE, Identities.ANONYMOUS, "");
this.asserts();
```

Reply:
@Vatavuk I agree that this looks confusing and I've decided to keep assertions in test methods and extract into method commands executions in test container, despite duplications it looks more readable and clearer